action/protocol/staking/handlers.go

```diff
@@ -663,19 +663,33 @@ func (p *Protocol) handleCandidateRegister(ctx context.Context, act *action.Cand
 		}
 	}
 
-	bucket := NewVoteBucket(owner, owner, act.Amount(), act.Duration(), blkCtx.BlockTimeStamp, act.AutoStake())
-	bucketIdx, err := csm.putBucketAndIndex(bucket)
-	if err != nil {
-		return log, nil, err
+	// register with self-stake
+	bucketIdx := uint64(candidateNoSelfStakeBucketIndex)
+	var bucketLogs []*action.TransactionLog
+	votes := big.NewInt(0)
+	if act.Amount().Sign() > 0 {
+		bucket := NewVoteBucket(owner, owner, act.Amount(), act.Duration(), blkCtx.BlockTimeStamp, act.AutoStake())
+		bktIdx, err := csm.putBucketAndIndex(bucket)
+		if err != nil {
+			return log, nil, err
+		}
+		bucketIdx = bktIdx
+		bucketLogs = append(bucketLogs, &action.TransactionLog{
+			Type:      iotextypes.TransactionLogType_CANDIDATE_SELF_STAKE,
+			Sender:    actCtx.Caller.String(),
+			Recipient: address.StakingBucketPoolAddr,
+			Amount:    act.Amount(),
+		})
+		votes = p.calculateVoteWeight(bucket, true)
+		log.AddTopics(byteutil.Uint64ToBytesBigEndian(bucketIdx), owner.Bytes())
```

Comment:
this needs to be outside of `if{}` -- registration without self-stake should also create a topic in the log